src/algebra/ordered_group.lean

```diff
@@ -831,13 +833,24 @@ lemma abs_abs_sub_abs_le_abs_sub (a b : α) : abs (abs a - abs b) ≤ abs (a - b
 abs_sub_le_iff.2 ⟨abs_sub_abs_le_abs_sub _ _, by rw abs_sub; apply abs_sub_abs_le_abs_sub⟩
 
 lemma abs_eq (hb : 0 ≤ b) : abs a = b ↔ a = b ∨ a = -b :=
-iff.intro
-  begin
-    cases le_total a 0 with a_nonpos a_nonneg,
-    { rw [abs_of_nonpos a_nonpos, neg_eq_iff_neg_eq, eq_comm], exact or.inr },
-    { rw [abs_of_nonneg a_nonneg, eq_comm], exact or.inl }
-  end
-  (by intro h; cases h; subst h; try { rw abs_neg }; exact abs_of_nonneg hb)
+begin
+  split,
+  { cases abs_choice a; rw h,
+    { exact or.inl },
+    { rw [neg_eq_iff_neg_eq, eq_comm], exact or.inr } },
+  { rintro (rfl|rfl); try { rw abs_neg }; exact abs_of_nonneg hb }
```

Comment:
```suggestion
  { rintro (h | h); simp only [h, abs_neg, abs_of_nonneg hb] },
```

Reply:
out of curiosity, what's the reason not to use `rfl`?

Reply:
`simp` on both goals would fail

Reply:
Seems to work fine, actually.

Reply:
(although this is no longer true the way is formulated above in response to your comment)

Reply:
Haha beat me to it, I noticed it worked in your most recent push.